purchase_receipt_percentage/models/purchase_order.py

```diff
@@ -0,0 +1,39 @@
+# Copyright 2023 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    receipt_percentage = fields.Float(
+        compute="_compute_receipt_percentage",
+        store=True,
+        help="Receipt percentage between 0% and 100%",
+    )
+    receipt_percentage_display = fields.Float(
+        compute="_compute_receipt_percentage",
+        store=True,
+        help="Technical field to be displayed in view. Its value is between"
+        " 0 and 1, the percentage widget will format the value properly",
+    )
```

Comment:
Why do you need two fields for the receipt percentage ?

Reply:
The first one stores the actual percentage (included between 0 and 100) and can be used in the code directly since it holds the real percentage, the other one should be used only in UI (and its value is included between 0 and 1, the `widget="percentage"` takes care of the proper conversion).